utils/controleval.py

```diff
@@ -49,95 +49,160 @@ def validate_args(ctrlmgr, args):
-        ctrlstats[controls.Status.PARTIAL]
-
-    print("Total controls = {total}".format(total=total))
-    print_specific_stat("Applicable", applicable, total)
-    print_specific_stat("Assessed", assessed, applicable)
-    print()
-    print_specific_stat("Automated",
-                        ctrlstats[controls.Status.AUTOMATED],
-                        applicable)
-    print_specific_stat("Manual",
-                        ctrlstats[controls.Status.MANUAL],
-                        applicable)
-    print_specific_stat("Supported",
-                        ctrlstats[controls.Status.SUPPORTED],
-                        applicable)
-    print_specific_stat("Documentation",
-                        ctrlstats[controls.Status.DOCUMENTATION],
-                        applicable)
-    print_specific_stat("Inherently Met",
-                        ctrlstats[controls.Status.INHERENTLY_MET],
-                        applicable)
-    print_specific_stat("Does Not Meet",
-                        ctrlstats[controls.Status.DOES_NOT_MEET],
-                        applicable)
-    print_specific_stat("Partial",
-                        ctrlstats[controls.Status.PARTIAL],
-                        applicable)
+    status_count = count_implicit_status(ctrls, status_count)
+    control_list = create_implicit_control_lists(ctrls, control_list)
 
-    applicablelist = ctrls - ctrllist[controls.Status.NOT_APPLICABLE]
-    assessedlist = set().union(ctrllist[controls.Status.AUTOMATED]).union(ctrllist[controls.Status.SUPPORTED])\
-        .union(ctrllist[controls.Status.DOCUMENTATION]).union(ctrllist[controls.Status.INHERENTLY_MET])\
-        .union(ctrllist[controls.Status.PARTIAL]).union(ctrllist[controls.Status.MANUAL])
-    print("Missing:", ", ".join(sorted(str(c.id)
-          for c in applicablelist - assessedlist)))
+    return status_count, control_list
+
+
+def print_specific_stat(status, current, total):
+    if current > 0:
+        print("{status:16} {current:6} / {total:3} = {percent:4}%".format(
+            status=status,
+            percent=round((current / total) * 100.00, 2),
+            current=current,
+            total=total))
 
 
-def print_specific_stat(stat, current, total):
-    print("{stat} = {percent}% -- {current} / {total}".format(
-        stat=stat,
-        percent=round((current / total) * 100.00, 2),
-        current=current,
-        total=total))
+def print_controls(status_count, control_list, args):
+    status = args.status
+    if status not in status_count:
+        print("Error: The informed status is not available")
+        print_options(status_count)
+        exit(1)
+
+    if status_count[status] > 0:
+        print("\nList of the {status} ({total}) controls:".format(total=status_count[status],
+                                                                status=status))
+        for ctrl in control_list[status]:
+            print("{id:>16} - {title}".format(id=ctrl.id, title=ctrl.title))
```

Comment:
It would be nice to sort the controls according to their IDs.

Reply:
Yes. That would be great and I even considered to do that. However, they are currently in a set, so I couldn't easily order them. But this is in my todo list to go deeper and improve in another PR.